src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java

```diff
@@ -122,9 +124,22 @@ private void proxyWmsRequest(HttpServletRequest request, HttpServletResponse res
 
             URL wmsUrl = new URL(wmsUrlStr + "?" + queryString);
 
-            IOUtils.copy(wmsUrl.openConnection().getInputStream(), response.getOutputStream());
+            HttpURLConnection connection = (HttpURLConnection)wmsUrl.openConnection();
+            if (connection.getResponseCode() != 200 ) {
+                String ret = String.format("ERROR proxying URL '%s' - %s", wmsUrl, connection.getResponseMessage());
+                response.sendError(connection.getResponseCode(), ret);
+                LOGGER.log(Level.SEVERE, ret);
+            }
+            else {
+                response.setStatus(connection.getResponseCode());
+                response.setContentType(connection.getContentType());
+                response.setContentLength(connection.getContentLength());
+                IOUtils.copy(connection.getInputStream(), response.getOutputStream());
```

Comment:
err, no.

Do it for the input stream.

https://docs.oracle.com/javase/tutorial/networking/urls/readingWriting.html